### DISCUSSION TITLE
Support on getting a battery model charge and discharge in Real-time

### DISCUSSION THREAD

**Discussion by francis-tj** [pybamm-team/PyBaMM#3764] [General]
I am new to PyBaMM and I have tried some of the examples too. I wonder if there is any possibility to run a battery model in real-time instead of getting instant results from it. For example, If I wish to run a charging emulation for 5 hours for a battery model, I want this model to run 5 hours straight and update the charging status every second or minute. This should go for the discharging setup also.
However from the example codes I have run so far, the model give me an instant result on how it may behave in that specified period of time.

Is there any way that I can use PyBaMM according to my requirement?

**Comment by TomTranter** [comment]
You can start a simulation from an existing solution so I think if you saved the solution every minute to a file and made the current an input you could run it in a loop triggering the next simulation when a new current file is added to a folder or something like that. That would be the simplest way to start. Ultimately you would probably want a rest api to pass in controls and update states
